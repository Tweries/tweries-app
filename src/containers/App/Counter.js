import React from 'react';

function Counter({ length }) {
  const value = length === 0 ? '' : length;
  return <small className="App__counter">{value}</small>;
}

export default Counter;
