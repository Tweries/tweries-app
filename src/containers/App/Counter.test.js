import { render } from '@testing-library/react';
import React from 'react';
import Counter from './Counter';

describe('Counter', () => {
  const scenarios = [
    {
      description: 'show false',
      props: {
        length: 123,
        show: false
      }
    },
    {
      description: 'show true, length 123',
      props: {
        length: 123,
        show: true
      }
    },
    {
      description: 'show true, length 0',
      props: {
        length: 0,
        show: true
      }
    }
  ];

  scenarios.forEach(({ description, props }) => {
    it(description, () => {
      const { container } = render(<Counter {...props} />);
      expect(container).toMatchSnapshot();
    });
  });
});
