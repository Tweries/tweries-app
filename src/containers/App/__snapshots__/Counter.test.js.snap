// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter show false 1`] = `
<div>
  <small
    class="App__counter"
  />
</div>
`;

exports[`Counter show true, length 0 1`] = `
<div>
  <small
    class="App__counter"
  />
</div>
`;

exports[`Counter show true, length 123 1`] = `
<div>
  <small
    class="App__counter"
  >
    123
  </small>
</div>
`;
